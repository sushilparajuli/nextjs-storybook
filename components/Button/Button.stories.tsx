import type { Meta, StoryObj } from "@storybook/react";
import { expect } from "@storybook/jest";
import { userEvent, within } from "@storybook/testing-library";

import { Button } from "./Button";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta: Meta<typeof Button> = {
  title: "Example/Button",
  component: Button,
  tags: ["docsPage"],
};

export default meta;
type Story = StoryObj<typeof Button>;

// More on writing stories with args: https://storybook.js.org/docs/7.0/react/writing-stories/args
export const Primary: Story = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    await userEvent.click(canvas.getByRole("button"));
    await expect(args.onClick).toHaveBeenCalled();
  },
  args: {
    color: "primary",
    children: "Button",
  },
};

export const Secondary: Story = {
  ...Primary,
  args: {
    color: "secondary",
    children: "Button",
  },
};

export const WarningButton: Story = {
  ...Primary,
  args: {
    color: "warning",
    children: "Button",
  },
};

export const DangerButton: Story = {
  ...Primary,
  args: {
    color: "danger",
    children: "Button",
  },
};
