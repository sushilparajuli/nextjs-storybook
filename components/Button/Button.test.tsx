import { render, screen } from "@/test-utils";
import userEvent from "@testing-library/user-event";
import { Button } from "./Button";
describe("Button test cases", () => {
  it("should render", () => {
    const onClick = jest.fn();
    const { asFragment } = render(<Button onClick={onClick}>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("check onClick  callback", () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Button</Button>);
    const element = screen.getByRole("button");
    userEvent.click(element);
    expect(onClick).toHaveBeenCalled();
  });
});
