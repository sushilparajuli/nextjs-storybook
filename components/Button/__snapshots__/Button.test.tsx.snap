// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases should render 1`] = `
<DocumentFragment>
  <button
    class="css-o0uw0a"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;
